networks/suzuka/suzuka-full-node/src/main.rs

```diff
@@ -14,16 +18,52 @@ async fn main() -> Result<(), anyhow::Error> {
 			.init();
 	}
 
-	let dot_movement = dot_movement::DotMovement::try_from_env()?;
-	let path = dot_movement.get_path().join("config.toml");
-	let config = suzuka_config::Config::try_from_toml_file(&path).unwrap_or_default();
-	let (executor, background_task) = SuzukaPartialNode::try_from_config(config)
-		.await
-		.context("Failed to create the executor")?;
+	// start signal handler to shutdown gracefully and call all destructor
+	// on program::exit() the destrutor are not called.
+	// End the program to shutdown gracefully when a signal is received.
+	let (stop_tx, mut stop_rx) = watch::channel(());
+	tokio::spawn({
+		let mut sigterm = signal(SignalKind::terminate()).context("Can't register to SIGTERM.")?;
+		let mut sigint = signal(SignalKind::interrupt()).context("Can't register to SIGKILL.")?;
+		async move {
+			loop {
+				select! {
+					_ = sigterm.recv() => println!("Receive SIGTERM"),
+					_ = sigint.recv() => println!("Receive SIGTERM"),
+				};
+				if let Err(err) = stop_tx.send(()) {
+					tracing::warn!("Can't update stop watch channel because :{err}");
+					return Err::<(), anyhow::Error>(anyhow::anyhow!(err));
+				}
+			}
+		}
+	});
 
-	tokio::spawn(background_task);
+	//Start suzuka node process
+	let (gb_jh, run_jh) = {
```

Comment:
The block seems unnecessary, but this is just stylistics.

Reply:
Since you won't be worry about merging on main until after this sprint, maybe we do want to spend a bit more time on style though.

Reply:
yes I remove it.